.travis.yml

```diff
@@ -17,95 +17,110 @@ install:
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.4 -Dscala=2.11 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_sql.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - source ./travis/build_and_run_all_spark-2.4_hadoop-2.7.sh
 
-    - name: spark-3.0_websearch
+    - name: spark-2.4_hadoop-2.7_websearch
       before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+        - source ./travis/export_env_spark_2.4_hadoop_2.7.sh
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.4 -Dscala=2.11 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_websearch.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - source ./travis/build_and_run_all_spark-2.4_hadoop-2.7.sh
 
-    - name: spark-3.0_ml
+    - name: spark-2.4_hadoop-2.7_ml
       before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+        - source ./travis/export_env_spark_2.4_hadoop_2.7.sh
       script:
-        - cp ./travis/benchmarks_ml.lst ./conf//benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.4 -Dscala=2.11 -Dhive=0.14 -Dhadoop=2.4
+        - cp ./travis/benchmarks_ml.lst ./conf/benchmarks.lst
+        - source ./travis/build_and_run_all_spark-2.4_hadoop-2.7.sh
 
-    - name: spark-3.0_graph
+    - name: spark-2.4_hadoop-2.7_graph
       before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+        - source ./travis/export_env_spark_2.4_hadoop_2.7.sh
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.4 -Dscala=2.11 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_graph.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - source ./travis/build_and_run_all_spark-2.4_hadoop-2.7.sh
 
-    - name: spark-2.4_micro
+    - name: spark-3.0_hadoop-3.2_micro
       before_script:
-        - ./travis/export_environment_variables_spark_2.4.sh
+        - source ./travis/export_env_spark_3.0_hadoop_3.2.sh
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12
         - cp ./travis/benchmarks_micro.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_2.4.sh
+        - source ./travis/build_and_run_all_spark-3.0_hadoop-3.2.sh
 
-    - name: spark-2.4_sql
+    - name: spark-3.0_hadoop-3.2_sql
       before_script:
-        - ./travis/export_environment_variables_spark_2.4.sh
+        - source ./travis/export_env_spark_3.0_hadoop_3.2.sh
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12
         - cp ./travis/benchmarks_sql.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_2.4.sh
+        - source ./travis/build_and_run_all_spark-3.0_hadoop-3.2.sh
```

Comment:
We don't need `source`, just execute it : )

Reply:
@gczsjdy  using "source ./travis/build_and_run_all_spark-3.0_hadoop-3.2.sh" will  start in current shell process , but using "./travis/build_and_run_all_spark_2.4.sh" will start in subshell process , if using the latter will cause hadoop to fail to start, because it can't find hadoop env.

Reply:
The failing reason may be you echoed `export xx` to a file, instead of directly export.

Reply:
@gczsjdy  Shouldn't be the reason. I used directly export at the beginning, but it didn't work well.

Reply:
https://unix.stackexchange.com/questions/312573/what-is-the-difference-between-and-source Let's use `. script` 

> `.` This also sources the script. The source command is not required by POSIX and therefore is less portable than the shorter ..